File: web/src/stores/patient.ts
import { create } from 'zustand'
import { devtools } from 'zustand/middleware'

interface Patient {
  id: string
  first_name: string
  last_name: string
  email?: string
  phone?: string
  date_of_birth?: string
  gender?: string
  address?: string
  medical_history?: string
  allergies?: string
  created_at: string
  updated_at: string
}

interface PatientState {
  patients: Patient[]
  currentPatient: Patient | null
  addPatient: (patient: Patient) => void
  updatePatient: (patient: Patient) => void
  deletePatient: (id: string) => void
  setCurrentPatient: (patient: Patient | null) => void
}

export const usePatientStore = create<PatientState>()(
  devtools((set) => ({
    patients: [],
    currentPatient: null,
    addPatient: (patient) => set((state) => ({ patients: [...state.patients, patient] })),
    updatePatient: (patient) =>
      set((state) => ({
        patients: state.patients.map((p) => (p.id === patient.id ? patient : p)),
      })),
    deletePatient: (id) => set((state) => ({ patients: state.patients.filter((p) => p.id !== id) })),
    setCurrentPatient: (patient) => set({ currentPatient: patient }),
  }))
)

File: ios/HCPilotApp/Models/Patient.swift
import Foundation

struct Patient: Identifiable, Hashable, Codable {
    let id: String
    let first_name: String
    let last_name: String
    let email: String?
    let phone: String?
    let address: String?
    let latitude: Double?
    let longitude: Double?
    let date_of_birth: String?
    let gender: String?
    let medical_history: String?
    let allergies: String?
    let archived_at: String?
    let created_at: String
    let updated_at: String?

    var isArchived: Bool { archived_at != nil }

    var full_name: String {
        "\(first_name) \(last_name)"
    }

    var initials: String {
        "\(first_name.prefix(1))\(last_name.prefix(1))"
    }
}
